### ISSUE TITLE
FrameRange's dynamic axis is inconsistent with matrix

### ISSUE THREAD

**Issue by prash1912** [Microsoft/CNTK#857]
Hiii everyone...I am trying to implement text classification using Bidirectional RnnLstm.

My code is :
# CNTK Configuration File for creating a slot tagger and an intent tagger.

command = TrainTagger:TestTagger

makeMode = false ; traceLevel = 2 ; deviceId = "-1"

rootDir = ".." ; dataDir  = "$rootDir$/formatted_data" ; modelDir = "$rootDir$/work"

modelPath = "$modelDir$/trial.cmf"

vocabSize = 1092 ; numLabels = 129 ; numIntents = 2    # number of words in vocab, slot labels, and intent labels

# The command to train the LSTM model
TrainTagger = {
    action = "train"
    BrainScriptNetworkBuilder = {
        inputDim  = $vocabSize$
        intentDim = $numIntents$
        embDim = 150
        hiddenDim = 150

        BiRecurrentLSTMLayer {outDim} = {
            F = RecurrentLSTMLayer {outDim, goBackwards=false}
            G = RecurrentLSTMLayer {outDim, goBackwards=true}
            apply (x) = Splice (BS.Sequences.Last(F(x)):BS.Sequences.First(G(x)))
        }.apply

        model = Sequential (
            EmbeddingLayer {embDim} :
            #BatchNormalizationLayer {normalizationTimeConstant=2048} :
            BiRecurrentLSTMLayer {hiddenDim} :
            #BatchNormalizationLayer {normalizationTimeConstant=2048} :
            DenseLayer {intentDim, initValueScale=7}
        )

        # features
        n = DynamicAxis()
        query        = Input {inputDim, dynamicAxis=n}
        intentLabels = Input {intentDim}

        # model application
        z = model (query)

        # loss and metric
        weightTable = ParameterTensor{(1:2),init='uniform'}
        weight = weightTable * intentLabels
        ce1  = ReduceLogSum (z) - TransposeTimes (intentLabels, z)
        ce   = ce1.* weight  # weighted
        errs1 = BS.Constants.One - TransposeTimes (intentLabels, Hardmax (z))  # same as ErrorPrediction()
        errs  = errs1 .* weight
        #ce   = CrossEntropyWithSoftmax (intentLabels, z)
        #errs = ErrorPrediction         (intentLabels, z)

        featureNodes    = (query)
        labelNodes      = (intentLabels)
        criterionNodes  = (ce)
        evaluationNodes = (errs)
        outputNodes     = (z)
    }

    SGD = {
        maxEpochs = 8 ; epochSize = 36000

        minibatchSize = 70

        learningRatesPerSample = 0.01*2:0.005*12:0.001
        gradUpdateType = "FSAdaGrad"
        gradientClippingWithTruncation = true ; clippingThresholdPerSample = 15.0

        firstMBsToShowResult = 10 ; numMBsToShowResult = 100
        parallelTrain = {
            parallelizationMethod = "DataParallelSGD"
            parallelizationStartEpoch = 2
            distributedMBReading = true
            dataParallelSGD = { gradientBits = 1 }
        }
    }

    reader = {
        readerType = "CNTKTextFormatReader"
        file = "$DataDir$/train_1.sparse"
        randomize = true
        input = {
            query        = { alias = "CW" ; dim = $vocabSize$ ;  format = "sparse" }
            intentLabels = { alias = "L" ; dim = $numIntents$ ; format = "sparse" }
            #slotLabels   = { alias = "S2" ; dim = $numLabels$ ;  format = "sparse" }
        }
    }
}

# Test the model's accuracy (as an error count)
TestTagger = {
    action = "eval"
    modelPath = $modelPath$
    reader = {
        readerType = "CNTKTextFormatReader"
        file = "$DataDir$/test_1.sparse"
        randomize = false
        input = {
            query        = { alias = "CW" ; dim = $vocabSize$ ;  format = "sparse" }
            intentLabels = { alias = "L" ; dim = $numIntents$ ; format = "sparse" }
            #slotLabels   = { alias = "S2" ; dim = $numLabels$ ;  format = "sparse" }
        }
    }
}

I am getting this Error:

EXCEPTION occurred: DataFor: FrameRange's dynamic axis is inconsistent with matrix: {numTimeSteps:58, numParallelSequences:1, sequences:[{seqId:0, s:0, begin:0, end:58}]} vs. {numTimeSteps:1, numParallelSequences:1, sequences:[{seqId:0, s:0, begin:0, end:1}]}

Snippet of My input text file format (train_1.sparse) is :
1       |CW 2:1 |# bos  |L 1:1
1       |CW 3:1 |# 12
1       |CW 4:1 |# disciples
1       |CW 5:1 |# the
1       |CW 6:1 |# new
1       |CW 7:1 |# testament
1       |CW 8:1 |# is
1       |CW 5:1 |# the
1       |CW 9:1 |# only
1       |CW 10:1        |# extensive
1       |CW 11:1        |# source
1       |CW 12:1        |# of
1       |CW 13:1        |# information
1       |CW 14:1        |# about
1       |CW 5:1 |# the
1       |CW 15:1        |# lives
1       |CW 12:1        |# of
1       |CW 5:1 |# the
1       |CW 3:1 |# 12
1       |CW 4:1 |# disciples
1       |CW 16:1        |# eos
1       |CW 2:1 |# bos  |L 1:1
1       |CW 17:1        |# throughout
1       |CW 5:1 |# the
1       |CW 18:1        |# bible
1       |CW 19:1        |# there
1       |CW 20:1        |# have
1       |CW 21:1        |# been
1       |CW 22:1        |# men
1       |CW 23:1        |# as
1       |CW 24:1        |# well
1       |CW 23:1        |# as
1       |CW 25:1        |# women
1       |CW 26:1        |# whom
1       |CW 27:1        |# god
1       |CW 28:1        |# has
1       |CW 29:1        |# chosen
1       |CW 30:1        |# to
1       |CW 31:1        |# accomplish
1       |CW 32:1        |# his
1       |CW 33:1        |# divine
1       |CW 34:1        |# plans
1       |CW 16:1        |# eos
1       |CW 2:1 |# bos  |L 1:1
1       |CW 35:1        |# yet
1       |CW 19:1        |# there

If i run same code for SLUHandsOn train data then it works fine as it is having 3 columns. How to handle above mentioned error. What does it mean?

**Comment by prash1912** [comment]
I observed log file which i created after posting this issue. That will help us to understand the problem.

Log contains:

Validating network, final pass.

WARNING: Node 'z.x.out' (RowStack operation): Dynamic axes mismatch between Node 'z.x._[0].out' (GatherPacked operation) and Node 'z.x._[1].out' (GatherPacked operation). If they are incompatible, this will fail later.
WARNING: Node 'ce1.MinusArgs[1]' (TransposeTimes operation): Dynamic axes mismatch between Node 'intentLabels' (InputValue operation) and Node 'z' (Plus operation). If they are incompatible, this will fail later.
WARNING: Node 'ce1' (Minus operation): Dynamic axes mismatch between Node 'ce1.MinusArgs[0].r' (ReduceElements operation) and Node 'ce1.MinusArgs[1]' (TransposeTimes operation). If they are incompatible, this will fail later.
WARNING: Node 'ce1' (Minus operation): Dynamic axes mismatch between Node 'ce1.MinusArgs[0].r' (ReduceElements operation) and Node 'ce1.MinusArgs[1]' (TransposeTimes operation). If they are incompatible, this will fail later.
WARNING: Node 'ce' (ElementTimes operation): Dynamic axes mismatch between Node 'ce1' (Minus operation) and Node 'weight' (Times operation). If they are incompatible, this will fail later.
WARNING: Node 'ce' (ElementTimes operation): Dynamic axes mismatch between Node 'ce1' (Minus operation) and Node 'weight' (Times operation). If they are incompatible, this will fail later.
WARNING: Node 'errs1.MinusArgs[1]' (TransposeTimes operation): Dynamic axes mismatch between Node 'intentLabels' (InputValue operation) and Node 'errs1.MinusArgs[1].rightMatrix' (Hardmax operation). If they are incompatible, this will fail later.

How to solve this issue??

**Comment by prash1912** [comment]
Sorry guys there was an error in my input file..
